Species organ tweaks

comma opened the pull request:
Adds ability to partally override internal organs list, there as similar for external.
Gets rid of bioprint products list, since it's been used for 1 (one) species, and is /already/ unmaintained.
Instead organ decides whether it can be printed or not, and bioprinter just gets all printable organs from species.

Visible effect is that you can now actually print stomaches


Also moves eye organ generation from head organ to eye organ, should be no visible changes for players, aside from lizard/skrell specific eyes descriptions happening now.

comma committed:
Species organ tweaks

Adds ability to partally override internal organs list, there as similar for external.
Gets rid of bioprint products list, since it's been used for 1 (one) species, and is /already/ unmaintained.
Instead organ decides whether it can be printed or not, and bioprinter just gets all printable organs from species.

comma committed:
Moves eye icon generation into eye icon from head

PurpleMartinJCK commented:
What's fallen behind on the bioprint organ list? Which PR make it out of date?

PurpleMartinJCK commented on code/modules/organs/external/species/nabber_stance.dm:
```diff
@@ -22,22 +19,3 @@
 		owner.pull_punches()
 		refresh_action_button()
-
-/obj/item/organ/external/head/nabber/get_eye_cache_key()
```
Somewhere in here, the changes break eye translucence and eye shield overlays.

comma commented:
Stomaches are missing from bio printer. Also butts

afterthought2 commented:
I'm unconvinced that this: properly checks for organs which can't be printed (all vital organs should not be printable, any external organ with the `CAN_HEAL_OVERKILL` flag should not be printable); doesn't print mechanical-only organs in the flesh printer; prints mechanical versions in the robo printer.

Also, @PurpleMartinJCK, can you see if your comment about the eyes has been resolved?

comma committed:
Tightened up printability checks

Now checking few more things aside from straight up NO PRINT THIS flag - vital organs, non-replacible stuff, robotic-from-start stuff.

comma commented:
>I'm unconvinced that this: properly checks for organs which can't be printed (all vital organs should not be printable, any external organ with the CAN_HEAL_OVERKILL flag should not be printable); 

Added checks for that

>doesn't print mechanical-only organs in the flesh printer; 

Added a check for that but it's unlikely you'd get DNA sample in first place

>prints mechanical versions in the robo printer.

robo printer uses a plain list of printable stuff so my changes don't really affect it
